pkg/util/catalog/app.go

```diff
@@ -0,0 +1,47 @@
+package catalog
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	catalogv1 "github.com/rancher/rancher/pkg/generated/controllers/catalog.cattle.io/v1"
+	"github.com/sirupsen/logrus"
+
+	"github.com/harvester/harvester/pkg/settings"
+)
+
+// FetchAppChartImage fetches image information from Spec.Values of Rancher RCD App.
+func FetchAppChartImage(appCache catalogv1.AppCache, namespace, name string, keyNames []string) (settings.Image, error) {
+	var image settings.Image
+
+	harvesterApp, err := appCache.Get(namespace, name)
+	if err != nil {
+		return image, fmt.Errorf("cannot get %s/%s app: %v", namespace, name, err)
+	}
+
+	var (
+		ok            bool
+		bytes         []byte
+		value         = harvesterApp.Spec.Chart.Values
+		previousValue map[string]interface{}
+	)
+
+	for _, key := range keyNames {
+		previousValue = value
+		if value, ok = value[key].(map[string]interface{}); !ok {
+			logrus.Debugf("cannot find key %s in layer(%s) %s/%s app chart value: %v", key, strings.Join(keyNames, ","), namespace, name, previousValue)
```

Comment:
maybe just saying `fail to get value of path %s' % strings.Join(keyNames, ".")` ?

FWIW, this does a similar thing: https://github.com/harvester/harvester/blob/aacd8a0a3a30a84f02e48a98d2f8b2d183a176bd/vendor/github.com/rancher/norman/types/values/values.go#L69-L85, but we can keep your implementation here.

Reply:
Okay, i think your message is more suitable.

I didn't even notice there is already a similar function to do this! I would like to use that to replace lol.